dan_layer/epoch_manager/src/base_layer/base_layer_epoch_manager.rs

```diff
@@ -194,11 +230,20 @@ impl<TAddr: NodeAddressable + DerivableFromPublicKey>
                 actual: registration.sidechain_id().map(|v| v.to_hex()),
             });
         }
-        let constants = self.get_base_layer_consensus_constants().await?;
-        let next_epoch = constants.height_to_epoch(block_height) + Epoch(1);
-        let next_epoch_height = constants.epoch_to_height(next_epoch);
+
+        let constants = self.base_layer_consensus_constants().await?;
+        let mut next_epoch = constants.height_to_epoch(block_height) + Epoch(1);
         let validator_node_expiry = constants.validator_node_registration_expiry;
 
+        // find the next available epoch
+        let mut next_epoch_vn_count = self.validator_nodes_count(next_epoch, registration.sidechain_id())?;
+        while next_epoch_vn_count == self.config.max_vns_per_epoch_activated {
```

Comment:
What if the count > `self.config.max_vns_per_epoch_activated`?

Reply:
in theory it should not be greater than maximum, but let me change it to `>=`

Reply:
Oh I see, this is counting the active set for each epoch.